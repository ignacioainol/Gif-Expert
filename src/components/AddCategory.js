import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(['']);

    const handleKeyPress = (e) => {
        setInputValue(e.target.value);

        console.log("handleInputChange llamado");
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleKeyPress}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};