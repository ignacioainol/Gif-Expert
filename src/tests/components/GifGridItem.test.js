import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = "Some Title";
    const url = 'https://localhost/image.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe de mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', () => {
        const paraph = wrapper.find('p');
        expect(paraph.text().trim()).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props ', () => {
        const img = wrapper.find('img');
        // console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('src')).toBe(url);
    });

    test('Debe de tener animate__fadeIn ', () => {
        const div = wrapper.find('div');
        // console.log(div.prop('className').includes('animate__fadeIn'));
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    })




})