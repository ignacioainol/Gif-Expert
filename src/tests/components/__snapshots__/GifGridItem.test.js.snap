// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Some Title"
    src="https://localhost/image.jpg"
  />
  <p>
    Some Title
  </p>
</div>
`;
